clients/js/test/extensions/interestBearingMint/updateInterestBearingMint.test.ts

```diff
@@ -0,0 +1,85 @@
+import test from 'ava';
+import {
+  createDefaultSolanaClient,
+  generateKeyPairSignerWithSol,
+  getCreateMintInstructions,
+  sendAndConfirmInstructions,
+} from '../../_setup';
+import { Account, generateKeyPairSigner, isSome, some } from '@solana/web3.js';
+import {
+  extension,
+  fetchMint,
+  getInitializeInterestBearingMintInstruction,
+  getUpdateRateInterestBearingMintInstruction,
+  Mint,
+} from '../../../src';
+
+test('it updates the interest bearing mint extension on a mint account', async (t) => {
+  const client = createDefaultSolanaClient();
+  const [rateAuthority, mint] = await Promise.all([
+    generateKeyPairSignerWithSol(client),
+    generateKeyPairSigner(),
+  ]);
+
+  // in bips
+  const rate = 10000;
+  const newRate = 20000;
+
+  // And a interest bearing pointer extension.
+  const interestBearingMintExtension = extension('InterestBearingConfig', {
+    rateAuthority: rateAuthority.address,
+    initializationTimestamp: BigInt(Math.floor(new Date().getTime() / 1000)),
+    lastUpdateTimestamp: BigInt(Math.floor(new Date().getTime() / 1000)),
+    preUpdateAverageRate: 10000,
+    currentRate: rate,
+  });
+
+  const [createMintInstruction, initMintInstruction] =
+    await getCreateMintInstructions({
+      authority: rateAuthority.address,
+      client,
+      extensions: [interestBearingMintExtension],
+      mint,
+      payer: rateAuthority,
+    });
+  await sendAndConfirmInstructions(client, rateAuthority, [
+    createMintInstruction,
+    getInitializeInterestBearingMintInstruction({
+      rateAuthority: rateAuthority.address,
+      mint: mint.address,
+      rate: rate,
+    }),
+    initMintInstruction,
+    getUpdateRateInterestBearingMintInstruction({
+      rateAuthority: rateAuthority,
+      mint: mint.address,
+      rate: newRate,
+    }),
+  ]);
```

Comment:
Would you mind doing this in two steps?

So first, you can use the `createMint` helper to set up the mint with the interest bearing extension (this will ensure you logic in `getInitializeInstructionsForExtensions` work). Then, you just send the update instruction as you're doing right now.

Reply:
Changed as review. Thanks for the review ❤️